creator-node/src/services/stateMachineManager/stateMachineUtils.js

```diff
@@ -126,7 +131,56 @@ const retrieveClockValueForUserFromReplica = async (replica, wallet) => {
   return clockValue
 }
 
+/**
+ * Returns an object that can be returned from any state machine job to record a histogram metric being observed.
+ * Example: to call histogram.observe('response_time', { code: '200' }, 1000), you would call this function with:
+ * makeHistogramToRecord('response_time', 1000, 'code', '200')
+ * @param {string} metricName the key of the name of the metric from prometheus.constants
+ * @param {number} metricValue the value to observe
+ * @param {string} [metricLabelName] the optional name of the label
+ * @param {string} [metricLabelValue] the optional value of the label
+ */
+const makeHistogramToRecord = (
+  metricName,
+  metricValue,
+  metricLabelName = '',
+  metricLabelValue = ''
+) => {
+  if (!Object.keys(MetricNames).includes(metricName)) {
+    throw new Error(
+      `Invalid metricName: ${metricName}. Options: ${JSON.stringify(
+        Object.keys(MetricNames)
+      )}`
+    )
+  }
+  if (typeof metricValue !== 'number') {
+    throw new Error(`Invalid non-numerical metricValue: ${metricValue}`)
+  }
+  const useMetricLabel = metricLabelName && metricLabelValue
+  if (
+    useMetricLabel &&
+    !Object.keys(MetricLabels[metricName])?.includes(metricLabelName)
+  ) {
+    throw new Error(`Invalid metricLabelName: ${metricLabelName}`)
+  }
+  if (
+    useMetricLabel &&
+    !MetricLabels[metricName][metricLabelName]?.includes(metricLabelValue)
+  ) {
+    throw new Error(`Invalid metricLabelValue: ${metricLabelValue}`)
+  }
+
+  const metric = {
+    metricName: MetricNames[metricName],
```

Comment:
why do this here instead of having the caller pass in this by reference e.g. https://github.com/AudiusProject/audius-protocol/blob/theo-add-sync-metric/creator-node/src/routes/tracks.js#L309

Reply:
nah i'm taling specifically about the `MetricNames[metricName]`, see my linked example for how it gets the name

Reply:
ah I see, this was because the MetricLabels uses the metric name key (e.g., 'ISSUE_SYNC_SECONDS_WAITING_FOR_COMPLETION_HISTOGRAM' instead of the value of the key in the mapping). related to your other comments, I changed it to pass around the constant var name rather than the string so it should be resolved

Reply:
cool yeah i can see how that's confusing